lib/rules/no-typos.js

```diff
@@ -149,6 +150,17 @@ module.exports = {
       });
     }
 
+    function reportErrorIfStaticLifecycleMethod(node) {
+      STATIC_LIFECYCLE_METHODS.forEach(method => {
+        if (method.toLowerCase() === node.key.name.toLowerCase() && !node.static) {
```

Comment:
this line doesn't match line 144 above; i think you need to also check that `method !== node.key.name`?

Reply:
Did you mean `method === node.key.name`?

Also, does it make sense to do both checks? What we want to do here is check if the current node is a static lifecycle method.

I'm using `.toLowerCase()` so that this is caught sooner by the user. For example, the following code would give two errors (note: I should add a test for this):

```js
class Foo extends Component {
  GetDerivedStateFromProps() {}
}
```

One for the typo and other for the missing `static` keyword.

If instead we do `method === node.key.name`, the user will first only see the typo error, and after fixing it would see the missing `static` keyword error.

Reply:
no, i mean `!==` - if the name is equal when lowercased, but *the same*, then it isn't a typo and shouldn't be reported.

Reply:
But I'm not reporting typos on the name here, I'm reporting missing `static` keywords.

Reply:
Yes, the other check covers that.